pkg/services/navtree/navtree.go

```diff
@@ -0,0 +1,602 @@
+package navtree
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/api/navlinks"
+	"github.com/grafana/grafana/pkg/infra/kvstore"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	ac "github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/apikey"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/org"
+	"github.com/grafana/grafana/pkg/services/pluginsettings"
+	pref "github.com/grafana/grafana/pkg/services/preference"
+	"github.com/grafana/grafana/pkg/services/star"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Service interface {
+	GetNavTree(c *models.ReqContext, hasEditPerm bool, prefs *pref.Preference) ([]*dtos.NavLink, error)
+}
+
+type ServiceImpl struct {
```

Comment:
Based on [Principles of how to structure a service in Grafana](https://github.com/grafana/grafana/blob/main/contribute/architecture/backend/package-hierarchy.md#principles-of-how-to-structure-a-service-in-grafana) would suggest putting implementation in a sub-package.

Reply:
done! moved some more dtos to navtree package